js/reference.js

```diff
@@ -3569,6 +3611,23 @@
                     newRef._displayname = otherFK.colset.columns[0].table.displayname;
                 }
 
+                // NOTE: this is for contextualized toComment and toCommentDisplay, to be implemented later
+                // display = otherFK.getDisplay(this._context);
+
+                // comment
+                if (otherFK.to_comment) {
```

Comment:
In this case, `to_comment` is already processed, so we don't need to do `_isValidModelComment`. But you should still do string type check.  The same comment is applicable to `from_comment` case below (an empty string is a valid value which is not handled here.)

Also in all the codes below that you're setting the value of `_commentDisplay`, you don't need any extra check and you should just use the value. With the current implementation (with the extra check), the `comment` might be coming from the source object, while the `commentDisplay` is derived from the `to_comment_display `.